airflow/opt/providers/etna/etna/hooks/etna.py

```diff
@@ -579,7 +580,7 @@ def upload_file(
                     "size was not specified, but file is streaming.  You must specify a size hint when using a stream."
                 )
 
-        if len(dest_path) > 1:
+        if len(dest_path) > 1 and not self.folder_exists(project_name, bucket_name, os.path.dirname(dest_path)):
```

Comment:
Tweaked this a bit, since it seemed like if someone tried uploading a file to Metis at an existing folder, the `self.create_folder` would throw an exception and derail the upload.

Reply:
Interesting, I actually figured create_folder gracefully handles existing folders?  But ok, it's all good.